accelerator/http.c

```diff
@@ -66,63 +66,61 @@ static status_t ta_get_url_parameter(char const *const url, int index,
   return SC_OK;
 }
 
-static status_t ta_http_process_request(ta_http_t *const http,
-                                        char const *const url,
-                                        char const *const payload,
-                                        char **const out) {
-  // TODO: TA api callback functions,
-  //      includes error response, handling option request
+static int set_response_content(status_t ret, char **json_result) {
+  int http_ret;
+  if (ret == SC_OK) {
+    return MHD_HTTP_OK;
+  }
+
+  cJSON *json_obj = cJSON_CreateObject();
+  if ((ret & SC_ERROR_MASK) == 0x03) {
+    http_ret = MHD_HTTP_NOT_FOUND;
+    cJSON_AddStringToObject(json_obj, "message", "Request not found");
+  } else if ((ret & SC_ERROR_MASK) == 0x07) {
+    http_ret = MHD_HTTP_BAD_REQUEST;
+    cJSON_AddStringToObject(json_obj, "message", "Invalid request header");
+  } else {
+    http_ret = MHD_HTTP_INTERNAL_SERVER_ERROR;
+    cJSON_AddStringToObject(json_obj, "message", "Internal service error");
+  }
+  *json_result = cJSON_PrintUnformatted(json_obj);
```

Comment:
I'm worry that `json_obj` may be leaked. Can you check it again for safety?

Reply:
You're right! `json_obj` should be freed here.